### DISCUSSION TITLE
avoid custom core-eval for contract deployment

### DISCUSSION THREAD

**Discussion by dckc** [Agoric/agoric-sdk#9717] [Ideas]
sketch:

in `fun1.contract.js`:

```js
export meta = {
  privateArgsShape: {
    storageNode: M.remote('published.fun1'); // or perhaps...
    storage: M.remote('StorageNode: published.fun1')
  }
}
```

a `deploy-contract.js` tool imports the contract, checks the meta, and builds core-eval code to fulfill it, either open-coded or by giving some data to an interpreter.

Some prototyping experience is available in...
 - [contractStarter.js](https://github.com/agoric-labs/ag-power-tools/blob/main/contract/src/contractStarter.js)

refs:
 - https://github.com/Agoric/agoric-sdk/issues/8594
 - https://github.com/Agoric/agoric-sdk/issues/8378

cc @Jovonni

**Comment by turadg** [comment]
https://github.com/Agoric/agoric-sdk/pull/9472 has an idea I played with to automate some naming

**Comment by dckc** [comment]
on July 17 , @Jovonni and I started experimenting...

 - https://github.com/Agoric/dapp-agoric-basics/pull/48

**Comment by dckc** [comment]
One of my experiments in this area was [blocked back on Sep 12](https://github.com/endojs/endo/issues/1741#issuecomment-2346972648). A PR just landed that addresses that issue.

**Comment by dckc** [comment]
a few weeks ago, I had another brainstorm:

 - https://github.com/Agoric/agoric-sdk/pull/10695
> Drive most of the core-eval code from `ContractMeta` and `permit`.

I'm not quite happy with where it ended up, but I think the general approach has promise.

cc @amessbee @anilhelvaci
